src/sparsezoo/api/utils.py

```diff
@@ -34,5 +34,20 @@ def to_snake_case(string: str):
 def map_keys(
     dictionary: Dict[str, str], mapper: Callable[[str], str]
 ) -> Dict[str, str]:
-    """Given a dictionary, update its key to a given mapper callable"""
-    return {mapper(key): value for key, value in dictionary.items()}
+    """
+    Given a dictionary, update its keys to a given mapper callable.
+        If the value of the dict is a List of Dict or Dict of Dict,
+        recursively map its keys
+    """
+    mapped_dict = {}
+    for key, value in dictionary.items():
+        if isinstance(value, List) or isinstance(value, Dict):
+            value_type = type(value)
+            mapped_dict[mapper(key)] = value_type(
+                map_keys(dictionary=sub_dict, mapper=to_snake_case)
```

Comment:
I would expect `mapper=mapper` here:
```suggestion
                map_keys(dictionary=sub_dict, mapper=mapper)
```

Reply:
I don't know that there's necessarily something to do about it... I guess you could add another `if` statement or something? but if type checking were being enforced, this call would be a type error. `field` is a `List[Any]` as far as I can tell, and yet `map_keys` is called as though all of the children of the list are of type `Dict`. I guess the type annotation for `dictionary` isn't right also? Doesn't seem like `Dict[str, str]` leaves any room for `Dict` or `List` to show up as fields.